src/run.ts

```diff
@@ -0,0 +1,343 @@
+
+    console.log(`You entered: '${answer}'`);
+
+    const clue = classicalGameClue(guess, secret);
+
+    console.log(`Clue : Hit: ${clue.hit} Blow: ${clue.blow}`);
+    if (clue.hit === 4) {
+      console.log('Congratulations! You guessed the secret!');
+      rl.close();
+    } else {
+      // Continue asking for the next guess
+      classicalGame();
+    }
+  });
+};
+
+/**
+ * A function to simulate the game with shuffled clues in the console.
+ */
+const shuffledClueGame = () => {
+  rl.question('Enter a guess (4 digits): ', (answer) => {
+    if (!/^\d{4}$/.test(answer)) {
+      console.log('You should enter exactly 4 digits!');
+      currentGame();
+      return;
+    }
+
+    const guess = answer.split('').map(Number);
+
+    console.log(`You entered: '${answer}'`);
+
+    const clue = shuffledGameClue(guess, secret);
+
+    console.log(`Clue is: ${clue.join(' ')}`);
+
+    // If all positions match, end the game
+    if (clue.join('') === '2222') {
+      console.log('Congratulations! You guessed the secret!');
+      rl.close();
+    } else {
+      // Continue asking for the next guess
+      shuffledClueGame();
+    }
+  });
+};
+
+/**
+ * A function to simulate the game with shifted clues in the console.
+ */
+const shiftedClueGame = () => {
+  rl.question('Enter a guess (4 digits): ', (answer) => {
+    if (!/^\d{4}$/.test(answer)) {
+      console.log('You should enter exactly 4 digits!');
+      currentGame();
+      return;
+    }
+    const guess = answer.split('').map(Number);
+
+    console.log(`You entered: '${answer}'`);
+
+    const clue = shiftedGameClue(guess, secret);
+
+    console.log(`Clue is: ${clue.join(' ')}`);
+    // If all positions match, end the game
+    if (clue.join('') === '2222') {
+      console.log('Congratulations! You guessed the secret!');
+      rl.close();
+    } else {
+      // Continue asking for the next guess
+      shiftedClueGame();
+    }
+  });
+};
+
+/**
+ * A function to simulate the game with current clue system in the console.
+ */
+const currentGame = () => {
+  rl.question('Enter a guess (4 digits): ', (answer) => {
+    if (!/^\d{4}$/.test(answer)) {
+      console.log('You should enter exactly 4 digits!');
+      currentGame();
+      return;
+    }
+
+    const guess = answer.split('').map(Number);
+
+    console.log(`You entered: '${answer}'`);
+
+    const clue = currentGameClue(guess, secret);
+
+    console.log(`Clue is: ${clue.join(' ')}`);
+
+    // If all positions match, end the game
+    if (clue.join('') === '2222') {
+      console.log('Congratulations! You guessed the secret!');
+      rl.close();
+    } else {
+      // Continue asking for the next guess
+      currentGame();
+    }
+  });
+};
+
+const secret: number[] = gameGuesses.totalAttempts[8];
+
+process.stdout.write('Start\n');
+
+// Comment out the games you don't want to play.
+currentGame();
+classicalGame();
+summedGame();
+shuffledClueGame();
+shiftedClueGame();
```

Comment:
```suggestion
function chooseGameMode() {
  console.log('1. Current Game Mode');
  console.log('2. Classical Game Mode');
  console.log('3. Summed Game Mode');
  console.log('4. Shuffled Game Mode');
  console.log('5. Shifted Game Mode');

  rl.question('Enter your choice (1-5): ', (mode) => {
    switch (mode) {
      case '1':
        currentGame();
        break;
      case '2':
        classicalGame();
        break;
      case '3':
        summedGame();
        break;
      case '4':
        shuffledClueGame();
        break;
      case '5':
        shiftedClueGame();
        break;
      default:
        console.log('Invalid choice. Please select a number between 1 and 5.');
        chooseGameMode();
    }
  });
}

chooseGameMode();
```
maybe we can make her more ux friendly like this, wdyt?

Reply:
Since its purpose was our trials, i didn't think like this, but it might help for our tests as well. Adding it.